experiments/AMIP/modular/coupler_driver_modular.jl

```diff
@@ -346,20 +362,21 @@ cs = CoupledSimulation{FT}(
 
 
 #=
-## Initial States Exchange
+## Initial Component Model Exchange
 =#
-## share states between models
-include("components/push_pull.jl")
-atmos_pull!(cs)
-parsed_args["ode_algo"] == "ARS343" ? step!(atmos_sim.integrator, Δt_cpl, true) : nothing
-atmos_push!(cs)
-land_pull!(cs)
+
+## share states and fluxes between models
+turbulent_fluxes = CombinedAtmosGrid()
+update_surface_fractions!(cs)
+import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes)
+update_sim!(cs.model_sims.atmos_sim, cs.fields, turbulent_fluxes) # would be good to rm dep in cs
+compute_combined_turbulent_fluxes!(cs.model_sims, cs.fields, turbulent_fluxes) # here computed using atmos functions
+parsed_args["ode_algo"] == "ARS343" ? step!(atmos_sim, Δt_cpl) : nothing
```

Comment:
Why do we have this extra `step!` here when using ARS343?

Reply:
I'm not sure about the details, but the `SciBase.step!` called via `DiffEq` initializes the atmos model variables internally (so we didn't need this when using `ODE.Rosenbrock23`), but `ClimaTimeSteppers` doesn't do the initialization. Dennis may know more details.

Reply:
@dennisYatunin could you expand on this? I'm curious if this is something we should have accounted for when switching ClimaLSM from using ODE to using CTS